plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java

```diff
@@ -182,6 +191,70 @@ static String bytesToStringConverter(Object value)
         return format("FROM_BASE64('%s')", Base64.getEncoder().encodeToString(slice.getBytes()));
     }
 
+    public static Field toField(String name, Type type)
+    {
+        if (type instanceof ArrayType) {
+            Type elementType = ((ArrayType) type).getElementType();
+            if (elementType instanceof RowType) {
+                return Field.newBuilder(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) elementType)).setMode(Field.Mode.REPEATED).build();
+            }
+            return Field.newBuilder(name, toStandardSqlTypeName(elementType)).setMode(Field.Mode.REPEATED).build();
+        }
+        if (type instanceof RowType) {
+            return Field.of(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) type));
+        }
+        return Field.of(name, toStandardSqlTypeName(type));
+    }
+
+    private static FieldList toFieldList(RowType rowType)
+    {
+        List<Field> fields = rowType.getFields().stream()
+                .map(field -> toField(field.getName().orElse("field" + rowType.getFields().indexOf(field)), field.getType()))
+                .collect(Collectors.toUnmodifiableList());
+        return FieldList.of(fields);
+    }
+
+    private static StandardSQLTypeName toStandardSqlTypeName(Type type)
```

Comment:
We will need to keep this in sync with the type mappings otherwise CTAS statement's schema might differ from the table that gets created.

Would it be helpful to define a bi-directional map somewhere with this mapping and use it for both read type mappings and write mappings?

Reply:
When we support CTAS, we will add `BigQueryOutputTableHandle` and the schema should be defined in same `BigQueryMetadata.createTable()`. So, such mismatch won't happen.